### ISSUE TITLE
Robux Shop

### ISSUE THREAD

**Issue by ExcellenceRBLX** [Pluviolithic/horror-simulator#47]
# Robux Shop

This will cover the Gamepass Frame, Scripting Robux Pets & its Frame, Fear/Gems Frames, and Scripting Boosts & its Frame. 

Script opening the RobuxShop UI [MainUI > RobuxShop] will open "RobuxShop" 
Script the Close Button in [RobuxShop > Close] 

## Gamepass Frame 
- [x] **Completed?**

### Opening The Gamepass Frame
This frame will be visible by default since it is the first page. The frame is located in [RobuxShop > Background > GamepassesFrame] 
There is a button on the left UI to open up each frame. The "Gamepasses" button in the RobuxShop UI will make this frame visible. 
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/8fab670c-12d8-457e-aee9-0df30ba57949)
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/70e512b9-ccf9-4c1b-9943-d128d8decbfe)

### Scripting The Gamepass Frame
*- The actual scrollingframe that contains each gamepass is in [RobuxShop > Background > GamepassesFrame > ScrollingFrame]*
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/b586c487-3b39-488c-b84d-6df7d51ef9ee)

- The name of each gamepass ImageButton corresponds to the name of each gamepass ID in the [Config >GamepassData > IDs] folder. That can be a way to identify each gamepass. 

- Purchase Button: [[Gamepass] > Purchase]  will prompt the purchase. 
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/b9c6d1b1-2d7c-4465-8d7e-39b2550e0fcc)

#### Gamepass Descriptions
*This frame will uniquely have a descriptions UI to display information about the gamepass they're hovering/clicking*

When they are hovering over a gamepass, the [GamepassesFrame > Description] ImageLabel will be visible. 
It will show the "DescriptionText" StringValue in each gamepass, in [GamepassesFrame > Description > TextLabel]
*If they are on mobile, if they click on the gamepass it'll stay on the screen for 10 seconds. Clicking on another gamepass will restart the cooldown of course.*
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/c016fd7f-7fae-46a1-8135-b8e33475053e)
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/0b2a08f9-d69c-4ce7-ae8a-d3d4aced2fdd)
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/07d22c43-a6f7-4467-8d9e-582da5eccdb7)

# Robux Pets/Pets Frame
- [x] **Completed?**

*Before we script the Pets Frame we should script the Robux Pets themselves first.*
### Robux Pets
*These pets are bought with Robux. Later when we add the rebirth update, they will not be deleted on rebirth. These pets will always be in the player's pet inventory and cannot be deleted (it will always be locked).*

- The two Robux Pets are the "Golden Dominus" and the "Reaper", and these pets are permanent. You can make a tag for these pets called "Undeletable" or "PermanentPet" or something, your choice.
Make sure pets under this tag can not be deleted.

- There is a physical place to buy these pets. The player can prompt to buy them with a proximity prompt located in the workspace
[BuyGoldenDominus > Prompt] buys the Golden Dominus pet.
[BuyReaper > Prompt] buys the Reaper pet. 
**The IDs for these pets are in [DevProductData > IDs]. **
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/92938999-e259-4e43-9f71-d25d45120474)
After they buy them the pet hatch screen pops up with an image of the pet, rarity, etc. and a copy of the pet will go into their inventory. This is a dev product so they can buy as many as they want.
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/340535b4-b4aa-4eb1-af0d-aa996385bf99)
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/800ef7b1-a631-4f5b-931b-98089ad208af)
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/fc818c90-792a-403b-bc20-b8ae081b8386)

## Pets Frame
- [x] **Completed?**

Players can also buy these pets in the Robux Shop. They will be able to buy 3 at once as well as explained below.

#### Opening The Frame
When the player presses [RobuxShop > Pets] on the left it will open the Pets frame, located in [RobuxShop > Background > PetsFrame]

#### Scripting The Frame
The name of each item in the PetsFrame corresponds to the IDs in [DevProductData > IDs] 
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/c48598cd-9da3-46d6-b2c0-dec65a229a3e)

The 2 new items allow the player to buy 3 pets at once. When they buy "3GoldenDominus" and "3Reaper" it will give them 3 of these at once. Their corresponding IDs are also in the DevProduct ID folder under the same names.
When they buy 3 pets at once, it will use the 3 pets hatch UI instead of the 1 hatch.
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/fd1fe669-9637-46d2-bd30-96309389ef1e)
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/8dbe07a5-e5ab-4197-8059-3d181a8c1ca1)

## Fear/Gems Frame
- [x] **Completed?**

In these frames players can buy fear and gems with dev products. These frames work a bit differently since it will give different amounts depending on what area they have unlocked, so there has to be a frame for each area. But the dev products are reused across each area.
*So example, if they've unlocked no areas, itll give 3 gems for the first product, and if they've unlocked Area 2 the frame will change to the 2nd area and will give 6 gems for the first product. *

*Read everything before starting, because I propose 2 ways to do this. The first way is what I fully instructed below. The second way will be briefly mentioned at the end.*
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/9e18110d-566c-4aee-9655-b4ba03d1d877)

### Fear Packs
Script the "Fear" button in the RobuxShop UI to open the [RobuxShop > Background > FearFrame]
Each Fear Pack Item are in each Area's frame (image below).
Lets first script purchasing the dev products. When the player presses Purchase, Each item in each Area's Frame corresponds to the IDs in [DevProductData > IDs], just like the other shop stuff.
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/c1d3ec60-6acd-4035-9ed5-ec0efc9ddcc8)
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/7146f03d-e404-46fd-8956-1cd9445aabdb)

The Frames in FearFrame will change depending on what areas they have unlocked/the area strength requirements. Each frame is named after its area.
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/6a1c8951-dc0c-4328-a7a5-d382c7b8e428)

Each item in each area's frame has an "Amount" Value which will be how much fear the player gives. Idk if this will work with you, so that is why I proposed a second idea further below to have everything be in the Config folder. But I much prefer this way.
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/1e26a495-d382-4623-952f-6a88e674a5ab)

### Gem Packs
Exactly like Fear Packs, but just gems instead.
Script the "Gems" button in the RobuxShop UI to open the [RobuxShop > Background > GemsFrame]
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/3423de42-4b5e-4f7e-9802-64cf75d7fc86)
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/99fc24c8-1259-46e2-a3d5-d0f2a744349c)

### Second Way
Another way we could do this is to have a gems and fear folder in IDs with the corresponding names.
And each Pack Value will have its ID, and will have "number" attributes under them that'll have the names of each area and the amount of gems it will give. You might like this way better.
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/968e8c9d-c58d-4a87-b980-a75c0418b3bc)
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/634f699f-162d-4d3a-b700-df1d080511e3)

We could also use tags some how. Just let me know what you do.

# Boosts
- [x] **Completed?**
Just like the gamepasses, boosts will be covered in a separate ticket. I don't want to make this ticket any longer than it should be. 
But you can make it so the UI can be opened:
[RobuxShop > Boosts] will open up [RobuxShop > Background > BoostsFrame] 
Once you script the open button you can check the completed box.
![image](https://github.com/Pluviolithic/horror-simulator/assets/99376966/e922cd00-e576-481c-945a-c5b3b0754dec)
